server/src/main/java/io/druid/server/coordinator/CoordinatorStats.java

```diff
@@ -20,63 +20,86 @@
 package io.druid.server.coordinator;
 
 import com.google.common.collect.Maps;
-import io.druid.collections.CountingMap;
+import it.unimi.dsi.fastutil.objects.Object2LongMap;
+import it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap;
 
+import java.util.Iterator;
 import java.util.Map;
-import java.util.concurrent.atomic.AtomicLong;
 
 /**
  */
 public class CoordinatorStats
 {
-  private final Map<String, CountingMap<String>> perTierStats;
-  private final CountingMap<String> globalStats;
+  private final Map<String, Object2LongOpenHashMap<String>> perTierStats;
```

Comment:
There are a few patterns of using `perTierStats` and `globalStats`:

 - Iterate tier values per key;
 - Get value by key and tier;
 - Check if the key is absent;
 - Get global stat by key;
 - Merge global stats;

I suggest to encapsulate those use-cases in methods of `CoordinatorStats` rather then exposing Maps. Iteration could be encapsulated similarly to the way suggested in `ServerManager`.